### PR TITLE
[QPD-389] Replace DB calls with API requests to ContexAlyze

### PR DESCRIPTION
**Pull request by shubham-quartic**
As DB queries needed to be removed from quartic_notebook, they had to be replaced by API calls and all the DB calls will be made in ContexAlyze instead where they can be made from synchronous context.

### CODE REVIEW

File: notebook/db_util.py [hunk 1]
```diff
@@ -1,20 +1,29 @@
...
 
 
-from deming import deming_setup
-
 # PG values.
 PG_DB_USER = os.environ.get('PG_DB_USER', '')
 PG_DB_NAME = os.environ.get('PG_DB_NAME', '')
 PG_DB_PASS = os.environ.get('PG_DB_PASS', '')
 PG_DB_HOST = os.environ.get('PG_DB_HOST', '')
 PG_DB_PORT = os.environ.get('PG_DB_PORT', '')
```
Comment (shivinquartic) [inline notebook/db_util.py]: Are these required?

Comment (shivinquartic) [reply notebook/db_util.py]: Data base setup and all is also not required I think you can remove that.

Comment (shubham-quartic) [reply notebook/db_util.py]: Yes, removed

File: notebook/db_util.py [hunk 2]
```diff
@@ -46,126 +37,112 @@
 
 django.setup()
```
Comment (shivinquartic) [inline notebook/db_util.py]: Is Django setup required?

Comment (shubham-quartic) [reply notebook/db_util.py]: Not required, removed it

**Review by shivinquartic — APPROVED** [review]
lgtm

**Review by shivinquartic — APPROVED** [review]
lgtm

**Review by tushar-quartic — APPROVED** [review]

File: notebook/db_util.py [hunk 3]
```diff
@@ -1,171 +1,125 @@
...
-        :return: List of Mlnode instances.
+        :return: A single JSON serialized MLNode instance
         """
-        kwargs = {column_name: column_value}
-        return self.get_active_devices(EdgeDevice.objects.filter(device_type__in=device_types.valid_ml_node_types,
-                                                                 system_status__is_connected=True).filter(**kwargs))
-
-    def delete_kernel_session(self, column_name, column_value):
+        ml_nodes = self.get_mlnodes()
+        return list(filter(lambda ml_node: ml_node[column_name] == column_value, ml_nodes))[0]
```
Comment (mprasoonquartic) [inline notebook/db_util.py:50]: use `next` directly. `next(filter(...),None)`

File: notebook/db_util.py [hunk 4]
```diff
@@ -1,171 +1,125 @@
...
+    def check_kernel_sessions(self, kernel_id: str) -> bool:
         """
         Check if there is an entry wih the following field and value.
-        :param column_name: Name of the Column. ( String )
-        :param column_value: Value of the Column. ( String )
+        :param kernel_id: UUID of the Kernel Session ( String )
         :return: Boolean Value.
         """
+        kernel_session = self.get_kernel_session(kernel_id)
+        return 'kernel_id' in kernel_session
```
Comment (mprasoonquartic) [inline notebook/db_util.py:93]: is `kernel_id` supposed to be string?

File: notebook/gateway/managers.py [hunk 5]
```diff
@@ -381,20 +381,18 @@ def _get_kernel_endpoint_url(self, kernel_id=None,
...
         if kernel_id:
-            kernel_session = self.db.get_kernel_session('kernel_id', kernel_id)
-            if kernel_session:
-                ml_node = kernel_session[0].ml_node
-                mlnode = self.db.get_mlnode_address_with_field('id', ml_node.id)
-                _url = f'{mlnode}/api/kernels'
-                _url = url_path_join(_url, url_escape(str(kernel_id)))
-
+            kernel_session = self.db.get_kernel_session(kernel_id)
+            ml_node_id = kernel_session["ml_node"]
```
Comment (mprasoonquartic) [inline notebook/gateway/managers.py:391]: no need to handle the `if` case?

File: notebook/db_util.py (hunk 4 above)
Comment (shubham-quartic) [reply notebook/db_util.py:93]: Just checking if in the JSON response we have got the field, can be handled better I will do these changes in a separate PR

**Pull request merged**